### ISSUE TITLE
Custom paths configuration do not work

### ISSUE THREAD

**Issue by ducreuxr** [jbake-org/jbake#643]
If I try to override the template folder's path in jbake.properties file with a relative or absolute path, running `jbake -b` command throws the following exception: 

```
JBake v2.6.5 (2020-04-22 22:45:29PM) [http://jbake.org]

Error: Required folder cannot be found! Expected to find [my-custom-template] at: /home/ducreuxr/Workspaces/blog/my-custom-template
org.jbake.app.JBakeException: Error: Required folder cannot be found! Expected to find [my-custom-template] at: /home/ducreuxr/Workspaces/blog/my-custom-template
	at org.jbake.app.configuration.JBakeConfigurationInspector.checkRequiredFolderExists(JBakeConfigurationInspector.java:67)
	at org.jbake.app.configuration.JBakeConfigurationInspector.ensureTemplateFolder(JBakeConfigurationInspector.java:40)
	at org.jbake.app.configuration.JBakeConfigurationInspector.inspect(JBakeConfigurationInspector.java:22)
	at org.jbake.app.UtensilsFactory.createDefaultUtensils(UtensilsFactory.java:19)
	at org.jbake.app.Oven.<init>(Oven.java:71)
	at org.jbake.launcher.Baker.bake(Baker.java:31)
	at org.jbake.launcher.Main.run(Main.java:97)
	at org.jbake.launcher.Main.run(Main.java:83)
	at org.jbake.launcher.Main.main(Main.java:56)
```

## Environment

Jbake version : 2.6.5
Java version : 1.8.0_252 (AdoptOpenJDK)

## Expected Behavior

The [documentation](https://jbake.org/docs/2.6.5/#default_template_directory) says:

> This option defines the folder that holds the template files for your project. **This can be a relative path from the root of your project or an absolute path reference.**
 
Eg: template.folder=templates

So, I created a "themes" folder and then add my custom theme folder inside like so :

`<my-project-root>/themes/my-custom-template`

then add the following property into jbake.properties file :

`template.folder=themes/my-custom-template`

## Current Behavior

If I run `jbake -b`, jbake expect to find my-custom-template folder into the root folder instead of themes/my-custom-template folder.

## Steps to Reproduce

1. Create a themes folder
2. Copy the "templates" folder into "themes" folder
3. Add the following property into jbake.properties file : template.folder=themes/templates
4. Run `jbake -b`

**Comment by jonbullock** [comment]
Thanks for taking the time to raise this.

**Comment by ancho** [comment]
ups. for the realtive paths I have a fix now. 
the absolute path situation needs a bit more work and cutsom path tests on windows.
does it really makes sense that assets, template and content folder is not relative to the specified source folder?

**Comment by ancho** [comment]
> does it really make sense that assets, template and content folder is not relative to the specified source folder?

Yes it does ;)

**Comment by jonbullock** [comment]
Current plan is to fix this via #644 in the v2.6.7 release.
